### ISSUE TITLE
Setups with Nullable types on strict mocks no longer match null

### ISSUE THREAD

**Issue by pkpjpm** [moq/moq4#184]
Here's an example of the problem:

```
        var mock = new Mock<ISimpleInterface>(MockBehavior.Strict);
        mock.Setup(x => x.SimpleMethod(It.IsAny<Guid?>()));
        mock.Object.SimpleMethod(null);
```

This used to work in Moq 4.2.1502.0911, but in 4.2.1506.2016 it fails. Workaround is to setup null explicitly instead of the Nullable Guid:

```
        mock.Setup(x => x.SimpleMethod(null));
```

**Comment by IharBury** [comment]
The problem is in https://github.com/Moq/moq4/commit/cc45976c8870322a170fc020a32d9191cc8d33f5 because of Nullable<T> is value type

**Comment by kzu** [comment]
So this should also be fixed by the merged PR #185. if so, plz close after verifying :)

**Comment by stakx** [comment]
The test at the top of this thread works fine in recent versions of Moq, so I am closing this.
